src/check_release.rs

```diff
@@ -150,6 +151,27 @@ fn print_triggered_lint(
                 Ok(())
             })?;
         }
+
+        if let Some(witness) = &semver_query.witness {
+            if witness_generation.show_hints {
+                let message = config
+                    .handlebars()
+                    .render_template(&witness.hint_template, &pretty_result)
+                    .context("Error instantiating witness hint template.")?;
+
+                config.log_info(|config| {
+                    let note = Style::new()
+                        .fg_color(Some(Color::Ansi(AnsiColor::Cyan)))
+                        .bold();
+                    writeln!(
+                        config.stdout(),
+                        "{note}note:{note:#} downstream code similar to the following would break:\n\
```

Comment:
here i just inlined the note code.  we can't have a method in `GlobalConfig` that takes an `impl Write` and renders a note/error/warning/etc., because `.stdout`/`.stderr` mutably borrows the whole `GlobalConfig` struct.  we could:
- make (associated) functions
- implement them on the streams themselves (using a newtype)
- revisit using a crate like `annotate-snippets` to render for us
- make macros

if we want a longer-term solution, if you think it is necessary.

Reply:
Revisiting with something like `annotate-snippets` down the line would be great. This is a reasonable solution for the meantime — it doesn't make sense to hold up witness generation over CLI formatting.